### ISSUE TITLE
Split Modula lexer into Modula-2 and Modula-3

### ISSUE THREAD

**Issue by 1997tn** [stefankueng/BowPad#308]
The two languages are different. Grouping them into the same Modula umbrella is not right.

**Comment by stefankueng** [comment]
since there's only one lexer for module (it mentions both modula-2 and modula-3):
https://github.com/ScintillaOrg/lexilla/blob/557d750299f2c27b0f35e219e686602eb5f6ff43/lexers/LexModula.cxx#L7

what exactly do you expect to be different?

I mean it's clear that different major versions of languages have differences. But what exactly is it that you think should be different **here**?

**Comment by 1997tn** [comment]
> since there's only one lexer for module (it mentions both modula-2 and modula-3): https://github.com/ScintillaOrg/lexilla/blob/557d750299f2c27b0f35e219e686602eb5f6ff43/lexers/LexModula.cxx#L7
> 
> what exactly do you expect to be different?
> 
> I mean it's clear that different major versions of languages have differences. But what exactly is it that you think should be different **here**?

Modula-2 and Modula-3 are two different languages, they use different source file extensions and are incompatible. Even if upstream grouped them into just one Modula lexer for convenient, we should use source file extensions detection to split the lexer name into Modula-2 lexer and Modula-3 lexer (underlying there is still only one Modula lexer).

**Comment by stefankueng** [comment]
But what should be different in respect to lexing? What is shown/colored wrong?

Please, give an example.

**Comment by 1997tn** [comment]
> But what should be different in respect to lexing? What is shown/colored wrong?
> 
> Please, give an example.

OK. My request to split it into two lexers on BowPad's lexers list is unreasonable. You win. But they are still different languages, using different source file extensions, very different to various Oberon dialects that still use the same source file extensions.

**Comment by stefankueng** [comment]
Yes, they are different. And the moon and the sun are two different objects.
Stating facts doesn't help.

I'm asking for the last time:
What exactly is wrong?

- what do you expect to happen (or be colored in a certain way for a specific file)?
- what happened instead?
